packages/repository-typeorm/src/repositories/typeorm-repository.ts

```diff
@@ -0,0 +1,291 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/repository-typeorm
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  EntityCrudRepository,
+  Entity,
+  DataObject,
+  Options,
+  Filter,
+  Where,
+  AnyObject,
+} from '@loopback/repository';
+import {
+  getRepository,
+  Repository,
+  SelectQueryBuilder,
+  QueryBuilder,
+  UpdateQueryBuilder,
+  DeleteQueryBuilder,
+} from 'typeorm';
+import {DeepPartial} from 'typeorm/common/DeepPartial';
+import {OrderByCondition} from 'typeorm/find-options/OrderByCondition';
+
+import {TypeORMDataSource} from './typeorm-datasource';
+
+import * as debugModule from 'debug';
+const debug = debugModule('loopback:repository:typeorm');
+
+/**
+ * An implementation of EntityCrudRepository using TypeORM
+ */
+export class TypeORMRepository<T extends Entity, ID>
+  implements EntityCrudRepository<T, ID> {
+  typeOrmRepo: Repository<T>;
+
+  constructor(
+    public dataSource: TypeORMDataSource,
+    public entityClass: typeof Entity & {prototype: T},
+  ) {}
+
+  private async init() {
+    if (this.typeOrmRepo != null) return;
+    this.typeOrmRepo = <Repository<T>>await this.dataSource.getRepository(
+      this.entityClass,
+    );
+  }
+
+  async save(entity: DataObject<T>, options?: Options): Promise<T | null> {
+    await this.init();
+    const result = await this.typeOrmRepo.save(<DeepPartial<T>>entity);
+    return <T>result;
+  }
+
+  async update(entity: DataObject<T>, options?: Options): Promise<boolean> {
+    await this.init();
+    await this.typeOrmRepo.updateById(entity.getId(), <DeepPartial<T>>entity);
```

Comment:
IMO, we should not need to change the type of `entity` here, we should change the API signature or `DataObject<T>` interface to support `DeepPartial<T>`.